### PR TITLE
Big ass rewrite

### PR DESCRIPTION
**Pull request by NeaGogu**

### CODE REVIEW

File: backend/internal/bumbal/bumbal_zones_helper.go [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+	return &plotModel, nil
+}
+
+// GetZoneListReponse gets the zones from bumbal using the jwt token
+// IMPORTANT: the jwt token must be valid
+func GetZoneListReponse(jwt string) (*BumbalZoneListResponse, error) {
+
+	// fail safe... I shot myself in the foot wit this one...
+	// update: I shot myself in the foot again, I was using TrimLeft ...
```
Comment (VictorVader) [inline backend/internal/bumbal/bumbal_zones_helper.go:86]: skill issue

File: backend/internal/bumbal/bumbal_zones_test.go [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+	"testing"
+)
+
+func TestGetZoneListReponse(t *testing.T) {
+
+	t.Run("Happy path", func(t *testing.T) {
+
+		t.Skip("Skipping test for now, need to mock the response from bumbal")
+
+		jwt := "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpYXQiOjE2NzkwNTA2MTYsImV4cCI6MTY3OTA3OTQxNiwiaXNzIjoic2VwMjAyMzAyLmJ1bWJhbC5ldSIsImxvZ2dlZF9vdXRfdXVpZCI6IjJkdmVzamh0LWk3M3ctcjhjOC1nNW4wLWMyOXNsNmE5M3hjZyIsInV1aWQiOiJqcXVid2oxeC1sdzZ2LXM1cjUtbDE5Ni10Y3JvNmNzMm13MGsiLCJ1c2VyX2lkIjoiMTEiLCJyb2xlX2lkIjoiNSIsImRldmljZV9pZCI6bnVsbCwidXNlcl91dWlkIjoiaXB3dnVjbzctangzbC05NTV3LWgzMXctMnZwZ2hjaW50NWE4IiwidXNlcl9wYXJ0eV9pZCI6IjEifQ.GkfKUWTPFN4_6ZEF7sviceF76iitgZt5vkqUjkA1qhM"
```
Comment (VictorVader) [inline backend/internal/bumbal/bumbal_zones_test.go:16]: sus

File: backend/internal/bumbal/bumbal_zones_test.go [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+	}
+
+}
+
+func TestFlowFromBumbalRequestToPlotModel(t *testing.T) {
+
+	// skip as GetZoneListReponse is not mocked and always calls bumbal's API
+	t.Skip()
+
+	jwt := "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpYXQiOjE2NzkyNTIzMDIsImV4cCI6MTY3OTI4MTEwMiwiaXNzIjoic2VwMjAyMzAyLmJ1bWJhbC5ldSIsImxvZ2dlZF9vdXRfdXVpZCI6IjkzM2sybGgwLTN0bmYtMzg3MS01NmY1LWxjNTk3dmE4NnRzOCIsInV1aWQiOiJqOGhwYXBrMC1vZm8wLTExYWItbjV4cS1zZjF1aHBzNzAwMDQiLCJ1c2VyX2lkIjoiMTEiLCJyb2xlX2lkIjoiNSIsImRldmljZV9pZCI6bnVsbCwidXNlcl91dWlkIjoiaXB3dnVjbzctangzbC05NTV3LWgzMXctMnZwZ2hjaW50NWE4IiwidXNlcl9wYXJ0eV9pZCI6IjEifQ.OEglHHbh9mNU0VSVgfU3Hca2gfiWLEabP9H_IXiQU1s"
```
Comment (VictorVader) [inline backend/internal/bumbal/bumbal_zones_test.go:149]: sus

**Review by VictorVader — DISMISSED** [review]

File: backend/README.md [hunk 4]
```diff
@@ -51,6 +51,71 @@ Normally you should not bother with anything else, you can just run the containe
...
+	"plot_created_at": "2023-03-21T20:19:04.518Z",
+	"plot_saved_at": "2023-03-21T20:19:04.518Z",
+	"origin": "bzone"
+}
+```
+- `[POST] /plot/save` -> used to save a new plot configuration. THIS DOES NOT UPDATE EXISTING PLOTS. The request body should have this structure:
+```json
+{
+	"plot_id": "blabla", <- not taken into account; can be ommited
+	"plot_name": "a new plot hi", <- requirerd!
```
Comment (Lucasmeteenc) [inline backend/README.md:102]: Please check spelling

File: backend/cmd/web/plotids_handler.go [hunk 5]
```diff
@@ -68,3 +76,177 @@ func (app *application) getUserPlotIDs(w http.ResponseWriter, r *http.Request) {
...
+
+func (app *application) SyncBumbalZones(w http.ResponseWriter, r *http.Request) {
+	// FIX: move this logic to a middleware
+	t := r.Header.Get("Authorization") // -> "Bearer <TOKEN>"
+
+	token := strings.TrimPrefix(t, "Bearer ")
+
+	// check if we indeed have a bearer token
+	if token == "" {
+		http.Error(w, "There is a problem with your token", http.StatusBadRequest)
```
Comment (Lucasmeteenc) [inline backend/cmd/web/plotids_handler.go:120]: make the error more descriptive

File: backend/internal/bumbal/bumbal_zones_helper.go [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+
+	"github.com/google/uuid"
+)
+
+const baseUrl = "https://sep202302.bumbal.eu/api/v2/"
+const (
+	zonesUrl = "zone"
+)
+
+func (b *BumbalZoneListResponse) ToRangeModelList() ([]models.ZoneModel, error) {
```
Comment (Lucasmeteenc) [inline backend/internal/bumbal/bumbal_zones_helper.go:22]: add comment that explains the function

File: backend/internal/bumbal/bumbal_zones_helper.go (hunk 1 above)
Comment (Lucasmeteenc) [reply backend/internal/bumbal/bumbal_zones_helper.go:86]: remove the comment

**Review by Lucasmeteenc — APPROVED** [review]
do your TODOs

File: backend/README.md (hunk 4 above)
Comment (NeaGogu) [reply backend/README.md:102]: no

File: backend/cmd/web/plotids_handler.go (hunk 5 above)
Comment (NeaGogu) [reply backend/cmd/web/plotids_handler.go:120]: no

File: backend/internal/bumbal/bumbal_zones_helper.go (hunk 6 above)
Comment (NeaGogu) [reply backend/internal/bumbal/bumbal_zones_helper.go:22]: no

**Pull request merged**